tests/unit/auth-login-form-spec.js

```diff
@@ -0,0 +1,231 @@
+                    AuthLoginFormFactory: this.AuthLoginFormFactoryMock,
+                });
+
+                inject(function (
+                    $compile,
+                    $rootScope,
+                    $templateCache,
+                    loginFormDirective,
+                    authLoginFormDirectiveFactory
+                ) {
+                    this.$compile = $compile;
+                    this.$rootScope = $rootScope;
+                    this.$templateCache = $templateCache;
+                    this.loginFormDirective = loginFormDirective[0];
+                    this.authLoginFormDirectiveFactory = authLoginFormDirectiveFactory;
+                });
+
+                this.$templateCache.put('templates/auth/login_form.html', '<p>Just testing</p>');
+            });
+
+            it('should be the same as authLoginFormDirectiveFactory', function () {
+                expect(this.loginFormDirective).toBe(this.authLoginFormDirectiveFactory);
+            });
+
+            it('should use the login_form template', function () {
+                expect(this.loginFormDirective.templateUrl).toBe('templates/auth/login_form.html');
+            });
+
+            it('should use AuthLoginFormFactory', function () {
+                this.$compile('<div login-form></div>')(this.$rootScope);
+                this.$rootScope.$apply();
+                expect(this.AuthLoginFormFactoryMock).toHaveBeenCalled();
+            });
+
+        });
+
+        describe('AuthLoginFormFactory', function () {
+
+            beforeEach(module('angular-token-auth.auth-login-form-factory'));
+
+            beforeEach(function () {
+                this.authModuleSettingsMock = {
+                    LOGIN_REDIRECT_URL: '/login-redirect',
+                };
+                module({
+                    authModuleSettings: this.authModuleSettingsMock,
+                });
+
+                inject(function (
+                    $location,
+                    authFactory,
+                    authActionsFactory
+                ) {
+                    this.$location = $location;
+                    this.authFactory = authFactory;
+                    this.authActionsFactory = authActionsFactory;
+                });
+            });
+
+            describe('when first injected', function () {
+
+                beforeEach(function () {
+                    spyOn(this.authFactory, 'getToken');
+                    spyOn(this.$location, 'url');
+                });
+
+                describe('with an auth token', function () {
+                    beforeEach(function () {
+                        this.authFactory.getToken.and.returnValue('abc123');
+                    });
+                    it('should redirect to LOGIN_REDIRECT_URL', function () {
+                        expect(this.$location.url).not.toHaveBeenCalled();
+                        inject(function (AuthLoginFormFactory) { // eslint-disable-line no-unused-vars
```

Comment:
It's an automatic login if an auth token has been saved: revisit the app and you'll be automatically logged back in.

When the `AuthLoginFormFactory` is first injected, the definition function is run and, as usual, Angular stores the return value as a singleton and doesn't run the definition again (thankfully). The "run once" bit is crucial here.

In the definition is [this bit of code](https://github.com/incuna/angular-token-auth/blob/9f766e81c2c42f5bc16170ef83f46fe89fd0ace2/src/providers/auth-login-form-factory.js#L27-L29) that runs at runtime. It means as soon as a thing injects `AuthLoginFormFactory`, the code checks for an auth token and redirects if it's there.

A better solution would be to have it as a method and call it in a `run` block, for the same effect. However that may be more difficult to test: not a reason to not do it, just a precaution. Or it could be left up to the project to call it.